library/Reporting/Hook/EmailProviderHook.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Icinga\Module\Reporting\Hook;
+
+use Icinga\Application\Hook;
+
+abstract class EmailProviderHook
+{
+    /**
+     * Get all Contact eMails
+     *
+     * @return mixed
+     */
+    abstract public function getContactEmails();
+
+    /**
+     * @return array
+     */
+    final public static function getProvider(): array
```

Comment:
I think this should be called `getProviders()` as it isn't just returning **one** provider.